src/facade/dragonfly_listener.cc

```diff
@@ -139,6 +142,18 @@ Listener::Listener(Protocol protocol, ServiceInterface* si) : service_(si), prot
   http_base_.reset(new HttpListener<>);
   http_base_->set_resource_prefix("http://static.dragonflydb.io/data-plane");
   si->ConfigureHttpHandlers(http_base_.get());
+
+  tcp_keepalive = absl::GetFlag(FLAGS_tcp_keepalive);
```

Comment:
This cache is not thread safe, in that an access to `tcp_keepalive` can occur in parallel to setting it.
However, are you sure it's something we need to cache? We read it upon receiving a new connection, it's definitely something that can be slowed down. I'd vote to reduce complexity and not cache.

Reply:
I concur. connection creation is not on a hotpath.

Reply:
So the reason here to use static variable is not to cache the flag but to enable the validation of the flag value i.e that it is bigger than 3. As when we call config set command we first set the flag value and then call the registered callback so I did the validation in the callback and if the validation passed I set the tcp_keepalive var. I spoke with @chakaz and he suggested to do this validation with using the absl custom flags inside the AbslParseFlag. I can do this but then the question raised do we really need this validation @romange ? I believe we have this validation today because we devide tcp_keepalive value by 3 when setting the interval between subsequent keep-alive probes. But I could just make it min value 1. Does this makes sense?